### PR TITLE
[ALLUXIO-1622] Remove UnderFileSystemCluster#readEOFReturnsNegative

### PR DESCRIPTION
**Pull request by leohjwang**

### CODE REVIEW

File: minicluster/src/main/java/alluxio/underfs/UnderFileSystemCluster.java [hunk 1]
```diff
@@ -113,14 +113,12 @@ public static UnderFileSystemCluster getUnderFilesystemCluster(String baseDir,
   }
 
   /**
-   * This method is only used by the alluxio.client.FileOutStreamIntegrationTest unit-test.
+   * Gets the UnderFileSystemClass.
```
Comment (apc999) [inline minicluster/src/main/java/alluxio/underfs/UnderFileSystemCluster.java]: for getters,  you can simply make it like 
```java
/** 
 * @return the {@link UnderFileSystem} class name
 */
```

Also in this codebase, we always trying to use `{@link}` when possible (to help potential IDE refactor)

File: minicluster/src/test/java/alluxio/underfs/UnderFileSystemClusterTest.java [hunk 2]
```diff
@@ -73,15 +73,21 @@ public void getTest() throws IOException {
   @Test
   public void readEOFReturnsNegativeTest() {
     Whitebox.setInternalState(UnderFileSystemCluster.class, "sUnderFSClass",
-            (String) null);
-    Assert.assertFalse(UnderFileSystemCluster.readEOFReturnsNegative());
+        (String) null);
+    String underFSClass = UnderFileSystemCluster.getUnderFSClass();
+    Assert.assertFalse(underFSClass != null
```
Comment (apc999) [inline minicluster/src/test/java/alluxio/underfs/UnderFileSystemClusterTest.java]: how about make this two asserts?
```java
Assert.assertNotNull(...)
Assert.assertEquals(...)
```

File: tests/src/test/java/alluxio/client/AbstractFileOutStreamIntegrationTest.java [hunk 3]
```diff
@@ -101,7 +101,10 @@ protected void checkWrite(AlluxioURI filePath, UnderStorageType underStorageType
 
       InputStream is = ufs.open(checkpointPath);
       byte[] res = new byte[(int) status.getLength()];
-      if (UnderFileSystemCluster.readEOFReturnsNegative() && 0 == res.length) {
+      String underFSClass = UnderFileSystemCluster.getUnderFSClass();
+      if (underFSClass != null
```
Comment (apc999) [inline tests/src/test/java/alluxio/client/AbstractFileOutStreamIntegrationTest.java]: `"alluxio.underfs.hdfs.LocalMiniDFSCluster".equals(underFSClass)`?

**Comment by apc999** [comment]
@leohjwang I left some minor comments.
Thanks for the patch.

**Comment by leohjwang** [comment]
@apc999 updated per your comments

**Comment by apc999** [comment]
+1 LGTM
thanks for the improvement!

**Pull request merged**